### PR TITLE
[model] support MiniCPM-V 4.0

### PR DESCRIPTION
**Pull request by tc-mb**
## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

File: python/sglang/srt/models/minicpmv.py [hunk 1]
```diff
@@ -773,8 +774,165 @@ def pad_input_ids(self, input_ids: List[int], image_inputs: MultimodalInputs):
...
+        super().__init__(config=config, quant_config=quant_config, prefix=prefix)
+        assert self.version == (4, 0)
+
+    def init_llm(
+        self,
+        config: LlamaConfig,
+        quant_config: Optional[QuantizationConfig] = None,
+        prefix: str = "",
+    ) -> nn.Module:
+        return LlamaForCausalLM(config=config, quant_config=quant_config, prefix=prefix)
```
[the remark this answers is not in the document]
Comment (tc-mb) [reply python/sglang/srt/models/minicpmv.py:833]: I have solved it, you can check it again.

**Comment by JustinTong0323** [comment]
Could you use this model to replace the old minicpmv test? Should be under test/srt/test_vision_xxxxx

File: python/sglang/srt/models/minicpmv.py [hunk 2]
```diff
@@ -773,8 +774,168 @@ def pad_input_ids(self, input_ids: List[int], image_inputs: MultimodalInputs):
...
+        vision_embedding = self.vpm(
+            pixel_values,
+            patch_attention_mask=patch_attn_mask,
+            tgt_sizes=tgt_sizes,
+        )
+        return vision_embedding
+
+    def get_image_feature(self, items: List[MultimodalDataItem]) -> torch.Tensor:
+        # list of tensors
+        pixel_values = flatten_nested_list([item.pixel_values for item in items])
```
Comment (JustinTong0323) [inline python/sglang/srt/models/minicpmv.py]: We have renamed `pixel_value` in `MultimodalDataItem` to `feature`

**Review by JustinTong0323 — CHANGES_REQUESTED** [review]
Thanks for you contribution! May you resolve the comments first?

**Comment by JustinTong0323** [comment]
Your version of sglang appears to be outdated. It is recommended that you attempt to merge the main branch first.

File: python/sglang/srt/models/minicpmv.py (hunk 2 above)
Comment (tc-mb) [reply python/sglang/srt/models/minicpmv.py]: ok

**Comment by tc-mb** [comment]
> Your version of sglang appears to be outdated. It is recommended that you attempt to merge the main branch first.

Your 3 revision suggestions have been modified. ^_^

**Review by JustinTong0323 — APPROVED** [review]
